src/apps/eos/updaters.py

```diff
@@ -0,0 +1,233 @@
+from trezor.messages import (
```

Comment:
@matejcik 
`def write_uint64_be(w: bytearray, n: int) -> int:`
how can we pass here hasher which is an object? `write_uint64_le(hasher, val)`
Maybe you meant `hasher.append(val)` instead of using `write_<any>`?

Reply:
this should work:
```
from utils import HashWriter
from trezorlib.crypto.hashlib import sha256

hasher = HashWriter(sha256())
write_uint64_be(hasher, val)
```

Reply:
...the `bytearray` type annotation on the `write_*` methods is misleading at best and wrong at worst, but `HashWriter`s are designed to work this way

Reply:
Removed updaters.py and all logic moved to corresponding writers. 
We now use only common writers for all hash append\extend operations. 

As for local bytearray variable, we need it, as EOS environment requires that we first hash length of the action and only then particular fields itself. So we either have to compute the length of the action beforehand, hash it and only then hash the fields one by one or do as its done - hash each field into new variable and then hash the length of the variable and then extend hasher with those bytes. It seems to be more convenient and clean approach, i mean introducing temporary variable. If you guys know some better approach, we will be pleased to change what we have now, as we are also not very happy with that.

Reply:
see how I did it here: https://github.com/keepkey/keepkey-firmware/blob/7a9db57cd4b15304b478443ea2b2ecdb46c51668/lib/firmware/eos-contracts/eosio.system.c#L85

Reply:
I don't love that I had to hardcode those constants, but splitting it into `8 + 8 + 16 + 16 + 1` makes it very explicit about how the particular constant was arrived at, and making testcases from transactions on-chain proves it's right.

Reply:
No, let's keep the bytearray for the sake of consistency. Some actions have a fixed length but others are dynamic, in which case you'd have to do trickery which is not worth the small memory saving.

BUT, the bytearray operation should not happen in `write_action_*`, because that's a whole load of unnecessary boilerplate. Instead, `process_action` should supply the bytearray, and finalize the hasher write.

Reply:
Should be resolved.